### PR TITLE
Test: Nigthly fixes

### PR DESCRIPTION
**Pull request by eloycoto**
- Fix issues on upgrade test
- Fix race conditions on policygen to avoid race conditions.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5131)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
\o/ Nightly Build is ok! 

```
15:01:44 [Nightly] SSSSSSSSSSS
15:01:44 [Nightly] Ran 196 of 295 Specs in 17284.320 seconds
15:01:44 [Nightly] SUCCESS! -- 196 Passed | 0 Failed | 0 Pending | 99 Skipped PASS
15:01:44 [Nightly] 
15:01:44 [Nightly] Ginkgo ran 1 suite in 4h48m6.262387046s
15:01:44 [Nightly] Test Suite Passed
```

https://jenkins.cilium.io/job/Cilium-PR-Nightly-Tests-All/173/execution/node/40/log/

File: test/k8sT/Nightly.go [hunk 1]
```diff
@@ -398,6 +398,7 @@ var _ = Describe("NightlyExamples", func() {
 		})
```
Comment (ianvernon) [inline test/k8sT/Nightly.go:398]: @eloycoto this commit needs a sign-off.

**Review by ianvernon — APPROVED** [review]
Only requested change is for a commit to get a sign-off - no actual code changes needed. Once the tests pass (or meet requirements you need for nightly), you can amend the commit and merge so you don't have to run the tests again.

**Comment by eloycoto** [comment]
As @ianvernon request, I sign the commit and it's ready to merge.

**Pull request merged**